File: MvvmJavaDemo/app/src/main/java/com/android/mvvmjava/view/ui/base/CoreFragment.java
package com.android.mvvmjava.view.ui.base;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import com.android.mvvmjava.MvvmApp;

abstract public class CoreFragment extends Fragment {

    protected MvvmApp mvvmApp;
    protected CoreActivity coreActivity;

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        mvvmApp = (MvvmApp) context.getApplicationContext();
        coreActivity = (CoreActivity) context;
    }
}

File: VideoGallery/app/src/main/java/com/android/videogallery/fragments/CoreFragment.kt
package com.android.videogallery.fragments

import android.content.Context
import androidx.fragment.app.Fragment
import com.android.VideoApp

abstract class CoreFragment : Fragment() {

    protected lateinit var videoApp:VideoApp

    override fun onAttach(context: Context) {
        super.onAttach(context)
        videoApp = context.applicationContext as VideoApp
    }
}